Plugin/Model/OrderRepositoryPlugin.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Ampersand\DisableStockReservation\Plugin\Model;
+
+use Magento\Sales\Model\OrderRepository;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\Data\OrderSearchResultInterface;
+use Magento\Sales\Api\Data\OrderExtensionFactory;
+use Ampersand\DisableStockReservation\Model\SourcesRepository;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterfaceFactory;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterface;
+
+/**
+ * Class OrderRepositoryPlugin
+ * @package Ampersand\DisableStockReservation\Plugin\Model
+ */
+class OrderRepositoryPlugin
+{
+    /**
+     * @var SourcesRepository
+     */
+    private $sourcesRepository;
+
+    /**
+     * @var OrderExtensionFactory
+     */
+    private $orderExtensionFactory;
+
+    /**
+     * @var SourceSelectionResultInterfaceFactory
+     */
+    private $sourceSelectionResultInterfaceFactory;
+
+    /**
+     * OrderRepositoryPlugin constructor.
+     * @param SourcesRepository $sourcesRepository
+     * @param OrderExtensionFactory $orderExtensionFactory
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultInterfaceFactory
+     */
+    public function __construct(
+        SourcesRepository $sourcesRepository,
+        OrderExtensionFactory $orderExtensionFactory,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultInterfaceFactory
+    ) {
+        $this->sourcesRepository = $sourcesRepository;
+        $this->orderExtensionFactory = $orderExtensionFactory;
+        $this->sourceSelectionResultInterfaceFactory = $sourceSelectionResultInterfaceFactory;
+    }
+
+    /**
+     * @param OrderRepository $subject
+     * @param OrderInterface $result
+     * @return OrderInterface
+     */
+    public function afterGet(OrderRepository $subject, OrderInterface $result): OrderInterface
+    {
+        return $this->applyExtensionAttributesToOrder($result, null);
+    }
+
+    /**
+     * @param OrderRepository $subject
+     * @param OrderSearchResultInterface $result
+     * @return OrderSearchResultInterface
+     */
+    public function afterGetList(OrderRepository $subject, OrderSearchResultInterface $result): OrderSearchResultInterface
+    {
+        $sourcesCollection = $this->sourcesRepository->getOrdersSourcesCollection($result);
+
+        foreach ($result->getItems() as $item) {
+            $sourceSelectionResult = $this->sourceSelectionResultInterfaceFactory->create([
+                'sourceItemSelections' => $sourcesCollection[$item->getEntityId()],
```

Comment:
I don't think we need to create a results object, we just need to set an array of the sources to our extension attribute